### PR TITLE
Fix all the things

### PR DESCRIPTION
**Pull request by kaylynn234**
This PR:

-Wraps markov code in an asynchronous executor so that it's not blocking. I'm slightly annoyed that you regenerate your model every markov call.
-Smooths out scatterbrain to use rewrite syntax and a comprehension.
-Probably something else I forgot to put here.

This PR definitely needs a revision/ second pair of eyes. I should not be writing code when I'm in this state. 
What I can test, I have. There may be some obscure issues issues stemming from the fact I don't yet have crimsoBOT's source burned into my skull.

Please don't stab me.

### CODE REVIEW

File: crimsobot/utils/markov.py [hunk 1]
```diff
@@ -164,3 +166,11 @@ def crimso() -> str:
...
 
     return output
+
+async def async_wrap(bot, func, *args, **kwargs):
+    """Wraps a sync function into an asynchronous executor. Useful everywhere but it's here just because."""
+
+    func = functools.partial(func, *args, **kwargs)
+    output = await bot.loop.run_in_executor(None, func)
+
+    return output
```
Comment (williammck) [inline crimsobot/utils/markov.py]: add a newline at the end of this file please

File: crimsobot/utils/markov.py [hunk 2]
```diff
@@ -152,9 +153,10 @@ def rovin() -> str:
 
 
 def crimso() -> str:
-    f = open(c.clib_path_join('text', 'crimso.txt'), encoding='utf8', errors='ignore')
-    text = f.read()
-    f.close()
+    """Generates crimsonic text."""
+
+    with open(c.clib_path_join('text', 'crimso.txt'), encoding='utf8', errors='ignore') as f:
+        text = f.read()
```
Comment (williammck) [inline crimsobot/utils/markov.py]: 👍

File: crimsobot/utils/markov.py [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
 import markovify
 import nltk
+import functools
```
Comment (williammck) [inline crimsobot/utils/markov.py]: this should go at the top with the rest of the stdlib imports

File: crimsobot/cogs/chat.py [hunk 4]
```diff
@@ -16,13 +16,12 @@ def __init__(self, bot: CrimsoBOT) -> None:
...
-
         # grab message contents (which are strings):
-        async for message in channel.history(limit=500):
-            if message.author.id != self.bot.user.id:
-                messages.append(message.content)
+        messages = [
+            message.content
+            async for message in ctx.channel.history(limit=500) 
+            if message.author.id != self.bot.user.id
+        ]
```
Comment (williammck) [inline crimsobot/cogs/chat.py]: Not a fan of this change. Is there a benefit to this that I'm not seeing? The list comprehension makes this less clear IMO.

File: crimsobot/cogs/chat.py [hunk 5]
```diff
@@ -126,7 +124,7 @@ def __init__(self, bot: CrimsoBOT) -> None:
         ]
 
         title = 'An excerpt from **THE FIRST NECROMANCER (sort of)**, by Monty Ward'
-        descr = m.rovin()
+        descr = await m.async_wrap(self, m.rovin)
```
Comment (williammck) [inline crimsobot/cogs/chat.py]: does this need to be `m.async_wrap(self.bot, m.rovin)`? the definition of `async_wrap` looks to want a Bot instance but you're giving it a Cog.

File: crimsobot/cogs/chat.py [hunk 6]
```diff
@@ -94,12 +93,11 @@ def __init__(self, bot: CrimsoBOT) -> None:
             ('Allen Crimsberg', 1950, 1997),
         ]
 
-        descr = m.poem(int(random.gauss(5, 1)))
+        descr = await m.async_wrap(self, m.poem, int(random.gauss(5, 1)))
```
Comment (williammck) [inline crimsobot/cogs/chat.py]: same as below comment.

File: crimsobot/cogs/chat.py (hunk 5 above)
Comment (kaylynn234) [reply crimsobot/cogs/chat.py]: should be self.bot, messed up instances

File: crimsobot/cogs/chat.py (hunk 4 above)
Comment (kaylynn234) [reply crimsobot/cogs/chat.py]: speed i believe, i personally find it more readable but it is up to taste

File: crimsobot/cogs/chat.py (hunk 6 above)
Comment (kaylynn234) [reply crimsobot/cogs/chat.py]: see previous

**Comment by kaylynn234** [comment]
i am bad at github so will close this and open one with these changes

**Pull request closed without merging**